### PR TITLE
Offline first

### PR DESCRIPTION
**Pull request by wgranger**
Describe your changes.
This PR allows the touch table to work offline. This will be marked WIP until https://github.com/zooniverse/panoptes-net-client/pull/9 is merged on the .NET client. Afterwards, the client will have to be updated on Nuget and the dependency updated here. 

Working towards #52 

# Review Checklist

- [x] Are tests passing?
- [x] Is the build free of output errors?

### CODE REVIEW

**Comment by wgranger** [comment]
## Update: May 23

This PR is now ready to review. This stayed WIP a bit longer than I wanted because the `ClassificationPanelViewModel` was being refactored to improve structure and remove unnecessary items. There's some more refactoring I'd like to do there, but I'd like some other PRs merged before making any more changes.

The biggest change to offline-first in the most recent commits improved how we handle GraphQL queries from Caesar. 

**Originally:** Each classifier (6, around the table) would load a subject queue (10 subjects) on app load. Caesar would then be queried for each new "CurrentSubject" at each classifier. The local DB would not be updated for each answer (smooth, features, star).

**Changes:** Now, classifiers do not begin with a subject queue. One is fetched only if a user chooses "random galaxy" after the first classification.

Each time a new space cutout is loaded in the center of the app (including the initial app load) Caesar is queried for each galaxy. If Caesar returns classification results, we can assume those results are accurate and update our local DB accordingly. 

If Caesar returns nothing, we assume there is no information or Caesar is down. Either way, the local DB is not updated.

On classification, we find a subject's classification counts from the local DB, increment the total count and answer selection, and update those results to the local DB.

**tl:dr**
- We use Caesar for reliable classification counts and local DB updating
- If Caesar is down, we make no update on subject load
- The local DB is updated with each classification

**Comment by wgranger** [comment]
Still WIP until the [.NET client PR](https://github.com/zooniverse/panoptes-net-client/pull/9) is merged

**Comment by wgranger** [comment]
I've updated the .NET client on Nuget, updated the dependencies here, and did a couple last minute checks. Should be good for a review.

**Comment by wgranger** [comment]
Rebased and ready for review.

**Comment by chelseatroy** [comment]
Hi Will!

Happy to review this. What steps would you recommend I follow to make sure it works? At what point should I turn off wifi, et cetera?

Thanks!
Chelsea

**Comment by wgranger** [comment]
I'm glad you're asking this. You can turn off wifi at any point, and the app should continue to work. The only issue is that subject images will contain a placeholder (the U!Scientist logo) there is no internet connection to download the image. However, when internet comes back, the images should download.

**Comment by chelseatroy** [comment]
Will look at this more again next week! The main thing, as you and I saw together, is that on first pass, I am not seeing a placeholder image for each of the galaxies. I'll look at this again on Monday, though, maybe it's some weird thing. Might be worth looking into though.

**Comment by wgranger** [comment]
Good call. I decided to remove the fallback image after a conversation in Slack. A fallback image isn't the best solution, because a user wouldn't be able to classify the image with a fallback icon. Instead, I'll push up a separate PR in the future that addresses this. 

The likely solution is that we will have the subject images accessible locally, so users are still able to see images even if the internet is down.
